lib/Models/BoxDrawing.ts

```diff
@@ -833,75 +861,94 @@ export default class BoxDrawing {
 
         moveStep = moveVector;
       } else if (this.keepHeightSteadyWhenMovingLaterally) {
-        // Move the box laterally on the globe while keeping the height
-        // steady. To do this we derive a new ellipsoid from the globe's
-        // ellipsoid by enlarging it so that the current box center falls on
-        // its surface. Then we find the intersection point of the camera ray
-        // through the latest mouse cursor position with this enlarged
-        // ellipsoid. The resulting point should be roughly the same height as
-        // the previous point and will also lie on the vector from the camera
-        // through the mouse cursor.
+        // Move the box laterally on the globe while keeping its height (almost) steady.
+        // To do this:
+        //   1. Find the exact point on the box surface mouse pick landed.
+        //   2. Derive a new ellipsoid such that the picked point on the box
+        //      lies on the new ellipsoid surface.
+        //   3. Find the point where the camera ray intersects this new
+        //      ellipsoid. This intersection point will have the same height as
+        //      that of the picked point while also visually coinciding with
+        //      the mouse cursor.
+        //   4. Use this intersection point to compute the box moveStep
```

Comment:
Previously, we were not considering the exact point where the user clicked on the box. This resulted in a jumpy start to the interaction when the box center aligns itself with the mouse cursor. Now we account for the clicked point which rectifies this issue.